packages/admin/admin-stories/src/docs/form/components/stories/FinalFormFields.stories.tsx

```diff
@@ -65,10 +65,21 @@ storiesOf("stories/form/FinalForm Fields", module)
             { value: "vanilla", label: "Vanilla" },
         ];
 
-        const initialValues = {
-            autocomplete: { value: "strawberry", label: "Strawberry" },
-            autocompleteAsync: { value: "strawberry", label: "Strawberry" },
-        };
+        const initialValues = React.useMemo(
+            // Why useMemo()?
+            // FinalForm reinitializes the form every time initalValues changes. A shallow equality check is used.
+            // Therefore, without useMemo() FinalForm would reinitialize on every render.
+            // for alternative approaches to prevent this behavior, see https://final-form.org/docs/react-final-form/types/FormProps#initialvaluesequal
```

Comment:
I had exactly the same problem today, and I used initialvaluesequal, I found that better, as it still works if the initialValues actually change (eg. due to another entry gets loaded).

The only reason for shallow equality is performance I guess?

(oh, and this problem only appears because you have nested object in this example)

Reply:
> I had exactly the same problem today, and I used initialvaluesequal, I found that better, as it still works if the initialValues actually change (eg. due to another entry gets loaded).

You are right, in a real life scenario initialvaluesequal would be the better solution (that's why I added a link to it). 

However, since this is an example, I fear that people are going to copy the `<FinalForm />` code below in their projects.
So, if we add a deep equality check in this example, people might copy it although it's not necessary in their use case.

Therefore, I decided to alter the `initialValues` in this example because people probably won't copy them.

Reply:
I've seen useMemo in several real world forms that load initialValues from an API.

And I would even say initialvaluesequal doesn't do much harm, or do you think it does?

Reply:
> And I would even say initialvaluesequal doesn't do much harm, or do you think it does?

No, in this case it doesn't. But for a deep equality check you need another dependency (like lodash), therefore I tried to avoid it in the example

Reply:
We could also move the `initialValues` initialization to the outside of the React component, as there's no dependency to a React state in this example. However, in most cases, the initial values will depend on data loaded from an API (and therefore require a React state), so `useMemo` is fine for me.

Reply:
> We could also move the `initialValues` initialization to the outside of the React component, as there's no dependency to a React state in this example.

When clicking on "Show code" in the docs, only the code inside the Story is shown. 
I wanted the initalValues to be visible, therefore I put them inside the component.